src/ring_theory/graded_algebra/basic.lean

```diff
@@ -77,4 +77,26 @@ def graded_algebra.decompose : A ≃ₐ[R] ⨁ i, 𝒜 i := alg_equiv.symm
   (graded_algebra.decompose 𝒜).symm (direct_sum.of _ i x) = x :=
 direct_sum.submodule_coe_alg_hom_of 𝒜 _ _
 
+/-- The projection maps of graded algebra-/
+def graded_algebra.proj (𝒜 : ι → submodule R A) [graded_algebra 𝒜] (i : ι) : A →ₗ[R] A :=
+(𝒜 i).subtype.comp $
+  (dfinsupp.lapply i).comp $
+  (graded_algebra.decompose 𝒜).to_alg_hom.to_linear_map
+
+lemma graded_algebra.proj_apply (i : ι) (r : A) :
```

Comment:
```suggestion
@[simp] lemma graded_algebra.proj_apply (i : ι) (r : A) :
```
I'd argue that `simp` should replace `proj` with `decompose` because the latter is an `alg_equiv` and so has more lemmas available.